File: scam-g18/src/main/java/es/codeurjc/scam_g18/dto/EventDTO.java
package es.codeurjc.scam_g18.dto;

import java.time.LocalDateTime;
import java.util.List;

import es.codeurjc.scam_g18.model.Status;

public record EventDTO(
        Long id,
        String title,
        String description,
        Double price,
        Integer priceCents,
        LocalDateTime startDate,
        LocalDateTime endDate,
        String startDateStr,
        String startTimeStr,
        String endDateStr,
        String endTimeStr,
        Integer capacity,
        String category,
        String locationName,
        String locationAddress,
        String locationCity,
        String locationCountry,
        Double locationLatitude,
        Double locationLongitude,
        List<String> speakerNames,
        List<String> sessionTimes,
        List<String> sessionTitles,
        List<String> sessionDescriptions,
        Status status
) {}

File: frontend/scamg-g18/app/dtos/EventDTO.ts
export interface EventDTO {
  id: number;
  title: string;
  description: string;
  price: number;
  priceCents: number;
  startDate: string;
  endDate: string;
  startDateStr: string;
  startTimeStr: string;
  endDateStr: string;
  endTimeStr: string;
  capacity: number;
  category: string;
  locationName: string;
  locationAddress: string;
  locationCity: string;
  locationCountry: string;
  locationLatitude: number;
  locationLongitude: number;
  speakerNames: string[];
  sessionTimes: string[];
  sessionTitles: string[];
  sessionDescriptions: string[];
  status: 'DRAFT' | 'PENDING_REVIEW' | 'PUBLISHED' | 'ARCHIVED';
}
